orthanc/orthanc-anon/plugin/pixl.py

```diff
@@ -219,42 +216,151 @@ def OnHeartBeat(output, uri, **request) -> Any:  # noqa: ARG001
     output.AnswerBuffer("OK\n", "text/plain")
 
 
-def ReceivedInstanceCallback(receivedDicom: bytes, origin: str) -> Any:
-    """Modifies a DICOM instance received by Orthanc and applies anonymisation."""
-    if origin == orthanc.InstanceOrigin.REST_API:
-        orthanc.LogWarning("DICOM instance received from the REST API")
-    elif origin == orthanc.InstanceOrigin.DICOM_PROTOCOL:
-        orthanc.LogWarning("DICOM instance received from the DICOM protocol")
-
-    # It's important that as much code in this handler as possible is inside this "try" block.
-    # This ensures we discard the image if anything goes wrong in the anonymisation process.
-    # If the handler raises an exception the pre-anon image will be kept.
-    try:
-        return _process_dicom_instance(receivedDicom)
-    except Exception:  # noqa: BLE001
-        orthanc.LogError("Failed to anonymize instance due to\n" + traceback.format_exc())
-        return orthanc.ReceivedInstanceAction.DISCARD, None
-
-
-def _process_dicom_instance(receivedDicom: bytes) -> tuple[orthanc.ReceivedInstanceAction, None]:
-    # Read the bytes as DICOM/
-    dataset = dcmread(BytesIO(receivedDicom))
+def _anonymise_dicom_instance(dataset: pydicom.Dataset) -> bytes:
+    """
+    Anonymise a DICOM instance.
 
-    # Attempt to anonymise and drop the study if any exceptions occur
+    Skip the instance for a study if a PIXLDiscardError is raised.
+    """
     try:
         study_identifiers = get_study_info(dataset)
         anonymise_and_validate_dicom(dataset, config_path=None, synchronise_pixl_db=True)
-        return orthanc.ReceivedInstanceAction.MODIFY, write_dataset_to_bytes(dataset)
+        return write_dataset_to_bytes(dataset)
     except PixlDiscardError as error:
-        logger.warning("Skipping {}: {}", study_identifiers, error)
-        return orthanc.ReceivedInstanceAction.DISCARD, None
-    except:  # noqa: E722 Allow bare except
-        # Called from a callback in orthanc so will never take down the service
-        # we want to do dicard anything on failure so we don't leak identifiable information
-        logger.exception("{} failed", study_identifiers)
-        return orthanc.ReceivedInstanceAction.DISCARD, None
+        logger.warning("Skipping instance for study {}: {}", study_identifiers, error)
```

Comment:
is `logger.warning` equivalent to `orthanc.LogWarning`?

Reply:
`orthanc.LogWarning` is controlled by orthanc and you can adjust the log level dynamically from the UI. I'd prefer using the logger as we can control the format of that.

I _think_ this anonymises the entire study so the warning message, function name and docstring etc. need updating? Probably also worth removing this except?

Reply:
I think this is only anonymising a single instance, as [in `_anonymise_study_instances` we iterate over the list of instances and pass in a single instance](https://github.com/SAFEHR-data/PIXL/blob/b6c7add0dd943db07d43902c6b7224720679d34f/orthanc/orthanc-anon/plugin/pixl.py#L322-L333) to this function, but I might be misunderstanding something